rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,159 @@
+
+>**📝Note:** This system will used across mobiles, tablets & desktops.
+
+### **But why 8pt grid?**
+There is no right or wrong answer here, but there are few reasons for choosing 8pt over all others (`5pt`, `6pt`, `7pt`...)
+- It scales perfectly in all type of screen displays, be it Android or iOS (`@0.75` or `@1.5`).
+- Apple & Google advice to use the same, without stating too many reasons though.
+- It's simple for us to do the math. The `4` & `8` are simpler to calculate and they are flexible enough to setup a scale.
+
+### **Base Unit**
+We'll start with the base unit, this will allow us to create a scale of all the sizes in your spatial system. For us this also means using multiples of `x8` to define dimensions, padding, and margin of elements.
+
+### **What system will work for Razorpay?**
+We should consider few cases that will make the system flexible for us. Few things to consider here are the **user needs**.
+
+As a merchant (user) do they want the interface to be spacious with large fonts and limited number of actions? Absolutely not, we need something that doesn't look *too-dense* or *too-spacious*. Therefore we'll consider **`4pt base unit`** which will give us the flexibility to scale the UI in multiples of `x4`.
+
+We also need to focus on the number of variables we'll have in our scale. These are the variables/tokens developers will use to while defining a grid layout.
+
+### **How to apply a spatial system?**
+Applying a system can come in the form of `padding`, `margins`, `width` & `height`. So there are two things we can control here *main sizing* & *internal element sizing*.
+
+- **Element-first (strict element sizing):** In this type the size takes the priority. For examples *buttons*, *form inputs*, *etc*.
+![Element-First Spacing](./images/spatial/element-first.png)
+
+- **Content first (strict internal padding):** This is to be used when the content is dynamic or less predictable. Therefore, we will enforce the strict internal padding to those components. Simply put, this is the spacing around the elements.
+![Content-First Spacing](./images/spatial/content-first.png)
+
+### **Grids**
+The grid system helps align page elements based on sequenced columns and rows. Basically it helps you arrange your structure into structured manner. Here we'll look into 2D organisation of information.
+
+### **Column Grid**
+As the name suggests, this type of grid will help you organise content into evenly spaced vertical columns. The spacer which divide this grid is called as `gutter`. We'll use a **12 column grid** as it helps divide the area into *halfs*, *thirds*, *fourths*, *sixths*.
+
+Here we'll have 3 types of column grids, these will be used depending on the use case.
+1. **Default:** This has a gutter of `32px` & will be used on the internal dashboard.
```

Comment:
in the image can you mark and highlight one of the gutter space with `32px`?

Reply:
Can you also mark the screen widths on which you have showed these grids? It'll be better to even see the pink bar widths in the image